www/src/pages/components/forms.js

```diff
@@ -330,6 +334,75 @@ export default withLayout(function FormControlsSection({ data }) {
       </p>
       <ReactPlayground codeText={RangeCustom} />
 
+      <LinkedHeading h="3" id="forms-custom-file">
+        File
+      </LinkedHeading>
+      <p>A custom styled File uploader.</p>
+      <Callout>
+        The custom <code>FormFile</code> will by default not visibly display
+        your selected file. This requires additional JS. The recommended plugin
+        to animate custom file input is{' '}
+        <a href="https://www.npmjs.com/package/bs-custom-file-input">
+          bs-custom-file-input
+        </a>
+        .
+      </Callout>
+      <ReactPlayground codeText={File} />
+
+      <h4>Translating or customizing the strings with HTML</h4>
+      <p>
+        Bootstrap also provides a way to translate the “Browse” text in HTML
+        with the <code>data-browse</code> attribute which can be added to the
+        custom input label (example in Dutch):
+      </p>
+      <Callout>
+        Note that the <code>data-browse</code> attribute does not to anything
+        unless the <code>custom</code> prop is set.
+      </Callout>
+      <ReactPlayground codeText={FileButtonTextHTML} />
+
+      <h4>Translating or customizing the strings with SCSS</h4>
+      <p>
+        Please refer to the official{' '}
+        <a href="https://getbootstrap.com/docs/4.4/components/forms/#translating-or-customizing-the-strings-with-scss">
+          Bootstrap documentation for translating via SCSS
+        </a>
+        . The <code>lang</code> prop can be used to pass the language.
+      </p>
+      <ReactPlayground codeText={FileButtonTextSCSS} />
+
+      <h4>Customizing FormFile rendering</h4>
+      <p>
+        When you need tighter control, or want to customize how the{' '}
+        <code>FormFile</code> component renders, it may be better to use it's
+        constituent parts directly.
+      </p>
+      <p>
+        By providing <code>children</code> to the <code>FormFile</code> you can
+        forgo the default rendering and handle it yourself. (You can still
+        provide an <code>id</code> to the <code>FormFile</code> and have it
+        propagate to the label and input).
+      </p>
+      <Callout>
+        <p>
+          When customizing the <code>FormFile</code> rendering it is important
+          to note the order of the <code>label</code> and <code>input</code>{' '}
+          elements.
+        </p>
+        <ul>
+          <li>
+            If you are not setting the <code>custom</code> prop the
+            <code>label</code> should be before the <code>input</code>.
+          </li>
+          <li>
+            If you are setting the custom prop the <code>input</code> element
+            has to be placed before the <code>label</code> or the{' '}
+            <code>buttonText</code> prop will not work.
+          </li>
```

Comment:
These sound like very specific rules that our end users can definitely get caught up in. Would it be better not to allow our end users to be able to set their own `FormFile` structure? I'm not sure how much customization they would be able to do here anyways.

Reply:
Well I have a specific use-case for myself actually. In a project using react-bootstrap.

We are looking to use FormFile when it is available. We will be using isValid and isInvalid. However we will also be adding "is-warning" through usage of classes and some SCSS.

And since the valid/invalid messages are displayed through some Bootstrap CSS magic you can extend that with your own variants. You do this by adding more states in their $form-validation-stages (https://github.com/twbs/bootstrap/blob/v4.4.1/scss/_variables.scss#L670)

Making it look like this:
```
$form-validation-states: map-merge(
  (
    "valid": (
      "color": $form-feedback-valid-color,
      "icon": $form-feedback-icon-valid
    ),
    "invalid": (
      "color": $form-feedback-invalid-color,
      "icon": $form-feedback-icon-invalid
    ),
    "warning": (
      "color": $form-feedback-warning-color,
      "icon": $form-feedback-icon-warning
    ),
  ),
  $form-validation-states
);
```
This will allow you to do like this, and this will require being able to customize the FormFile markup. Of course the "is-warning" class is managed through a function to mimic isValid/isInvalid that exists on the component already.
```
<Form.File id="formcheck-api-custom" custom>
	<Form.File.Input className="is-warning" />
	<Form.File.Label data-browse="Button text">
		Custom file input
	</Form.File.Label>
	<Form.Control.Feedback className="warning-feedback">You did it!</Form.Control.Feedback>
</Form.File>
```

Reply:
I have expanded the docs for Form.File in the API section to make this even clearer
![Screenshot 2020-03-13 at 09 30 23](https://user-images.githubusercontent.com/322759/76603486-5566c480-650d-11ea-8eda-f77d2ee2c4e2.png)

Do you think this will be enough @bpas247 ?

Reply:
Yeah that's probably fine then 😃 great work!